docs/src/docs/asciidoc/guides/couchbase.adoc

```diff
@@ -0,0 +1,133 @@
+= Spring Session with Couchbase
+Mariusz Kopylec
+:toc:
+
+This guide describes how to use Spring Session backed by Couchbase.
+
+== Updating Dependencies
+Before you use Spring Session, you must ensure to update your dependencies.
+We assume you are working with a working Spring Boot web application.
+If you are using Gradle, ensure to add the following dependencies:
+
+.gradle.build
+[source,groovy]
+[subs="verbatim,attributes"]
+----
+dependencies {
+    compile group: 'org.springframework.data', name: 'spring-data-couchbase', version: '+'
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: '+'
+}
+----
+
+ifeval::["{version-snapshot}" == "true"]
+Since We are using a SNAPSHOT version, we need to ensure to add the Spring Snapshot Maven Repository.
+Ensure you have the following in your build.gradle:
+
+.gradle.build
+[source,groovy]
+----
+repositories {
+    maven { url 'https://repo.spring.io/libs-snapshot' }
+}
+----
+endif::[]
+
+ifeval::["{version-milestone}" == "true"]
+Since We are using a Milestone version, we need to ensure to add the Spring Milestone Maven Repository.
+Ensure you have the following in your build.gradle:
+
+.gradle.build
+[source,groovy]
+----
+repositories {
+    maven { url 'https://repo.spring.io/libs-milestone' }
+}
+----
+endif::[]
+
+[[how-to-use]]
+== How to use
+
+**Basic usage:**
+
+Configure Couchbase connection using Spring Data Couchbase. You can find a minimal configuration example link:http://projects.spring.io/spring-data-couchbase/#quick-start[here].
+
+Enable Couchbase backed HTTP session using `@EnableCouchbaseHttpSession`:
+
+[source,java]
+----
+@EnableCouchbaseHttpSession(namespace = "my-application")
+@SpringBootApplication
+public class Application {
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+}
+----
+
+Simply use `HttpSession` interface to control HTTP session. For example:
+
+[source,java]
+----
+@Controller
+public class SessionController {
+
+    @RequestMapping("uri")
+    public void doSomething(HttpSession session) {
+        ...
+    }
+}
+----
+
+**Namespaces support:**
```

Comment:
Generally this is for microservices architecture systems. There couple of reasons I decided to go with namespace:
1. Session data in a large distributed system (aspecially in microservices architecture) can be pretty big. Having app and global namespaces reduces the amount of data to read every `HttpSession.getSession()` invocation. You can gain a significant performance boost using session namespaces.
2. There is alomst none additional complexity for a developer. If you don't want to use them you can set the same namespace in all your apps in the system or use only global namespace. Maybe by default namespaces support could be turned off.
3. With namespaces multiple different microservices in teh system can set or get a session attribute of the same without conflicting each other.